### DISCUSSION TITLE
Add the maven scope in Component object ?

### DISCUSSION THREAD

**Discussion by FloJit** [CycloneDX/cyclonedx-maven-plugin#534] [Ideas]
Hi all,

this plugin is very helpfull to get dependecies of a product during a maven build.
But the scope value of a Component doesn't really match (in our case) with the reality. For example if we add the maven test scope in the configuration, mockito will be added as a required component, which is not the case to use our artifact (we don't need to deliver it).
Then actually, to seperate each scope I run 5 times the goal with each time only one scope to true (but it means to have 5 boms).
Well could it be possible to add a child in Component like a "mavenScope" to get the source scope of a dependency ?

Thanks.
Florent

**Comment by skhokhlov** [comment]
It's similar to this issue https://github.com/CycloneDX/cyclonedx-maven-plugin/issues/505

**Reply by FloJit** [reply]
Yes thanks.
